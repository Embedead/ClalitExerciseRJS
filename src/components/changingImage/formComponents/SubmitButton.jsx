import React from "react";
import styled from "styled-components";

const Submit = styled.button`
  margin: 0.5rem;
  background-color: ${(props) => props.color};
  border-radius: 6px;
  font-size: 18px;
  border-style: none;
  padding: 0.5rem;
  cursor: pointer;
  box-shadow: 0px 4px 8px 8px rgba(0, 0, 0, 0.05);
  &:hover {
    background-color: black;
    color: white;
  }
`;

export const SubmitButton = ({ children, color = "aquamarine" }) => {
  return <Submit color={color}>{children}</Submit>;
};
