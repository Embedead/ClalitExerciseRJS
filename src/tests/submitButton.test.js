import "@testing-library/jest-dom/extend-expect";
import {
  fireEvent,
  render,
  cleanup,
  screen,
  waitFor,
} from "@testing-library/react";
import React from "react";
import { SubmitButton } from "../components/changingImage/formComponents/SubmitButton";

test("button props change color", async () => {
  render(<SubmitButton color="green">SUBMIT</SubmitButton>);
  const button = screen.getByText("SUBMIT");
  // fireEvent.mouseOver(button);
  expect(button).toMatchSnapshot();
});
