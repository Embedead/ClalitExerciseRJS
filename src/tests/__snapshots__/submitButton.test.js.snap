// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button props change color 1`] = `
<button
  class="sc-bdnxRM jJUQsV"
  color="green"
>
  SUBMIT
</button>
`;
